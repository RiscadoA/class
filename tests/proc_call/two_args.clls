proc wait_and_close(x:wait, y:close) {
    println("wait x");
    wait x;
    println("close y");
    close y
};;

proc main() {
    cut {
        println("close x");
        close x
        |x: wait|
        cut {
            wait_and_close(x, y)
            |y:wait|
            println("wait y");
            wait y;
            ()
        }
    }
};;
