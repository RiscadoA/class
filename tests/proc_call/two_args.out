close x
wait x
close y
wait y
